From bc46143992955f1449e283842bb3b700d04ff1a3 Mon Sep 17 00:00:00 2001
From: Ricardo Neri <ricardo.neri@ti.com>
Date: Mon, 1 Nov 2010 10:40:34 -0600
Subject: ASoC: ABE-DSP: Correct gains of equalizers

Correct values of equalizer profiles to express gains in dB.
Values are converted using the formula gainDb = 20*log10(gainLinear)

Change-Id: Ia01684c72acda7248561e544a4a9c96ccb2ce00c
Signed-off-by: Ricardo Neri <ricardo.neri@ti.com>
---
 sound/soc/omap/omap-abe-coef.h |   44 ++++++++++++++++++++--------------------
 sound/soc/omap/omap-abe-dsp.c  |   36 ++++++++++++++++----------------
 2 files changed, 40 insertions(+), 40 deletions(-)

Index: Kernel/sound/soc/omap/omap-abe-coef.h
===================================================================
--- Kernel.orig/sound/soc/omap/omap-abe-coef.h	2011-01-19 00:10:05.102071740 +0530
+++ Kernel/sound/soc/omap/omap-abe-coef.h	2011-01-19 00:19:28.592697384 +0530
@@ -47,24 +47,24 @@
  * Coefficients for DL1EQ
  */
 const s32 dl1_equ_coeffs[NBDL1EQ_PROFILES][NBDL1COEFFS] = {
-/* Flat response with Gain =1 */
+/* Flat response with Gain = 0dB */
 				{0, 0, 0, 0, 0,
 				0, 0, 0, 0, 0,
 				0, 0, 0x040002, 0, 0,
 				0, 0, 0, 0, 0,
 				0, 0, 0, 0, 0},
 
-/* 800Hz cut-off frequency and Gain = 1  */
+/* 800Hz cut-off frequency and Gain = 0dB  */
 				{0, 0, 0, 0, 0, 0, 0, 0, 0, -7554223,
 				708210, -708206, 7554225, 0, 0, 0, 0,
 				0, 0, 0, 0, 0, 6802833, -682266, 731554},
 
-/* 800Hz cut-off frequency and Gain = 0.25 */
+/* 800Hz cut-off frequency and Gain = -12dB */
 				{0, 0, 0, 0, 0, 0, 0, 0, 0, -3777112,
 				5665669, -5665667, 3777112, 0, 0, 0, 0,
 				0, 0, 0, 0, 0, 6802833, -682266, 731554},
 
-/* 800Hz cut-off frequency and Gain = 0.1 */
+/* 800Hz cut-off frequency and Gain = -20dB */
 				{0, 0, 0, 0, 0, 0, 0, 0, 0, -1510844,
 				4532536, -4532536, 1510844, 0, 0, 0, 0,
 				0, 0, 0, 0, 0, 6802833, -682266, 731554},
@@ -74,24 +74,24 @@
  * Coefficients for DL2EQ_L
  */
 const s32 dl20_equ_coeffs[NBDL20EQ_PROFILES][NBDL2COEFFS] = {
-/* Flat response with Gain =1 */
+/* Flat response with Gain = 0dB */
 				{0, 0, 0, 0, 0,
 				0, 0, 0, 0, 0,
 				0, 0, 0x040002, 0, 0,
 				0, 0, 0, 0, 0,
 				0, 0, 0, 0, 0},
 
-/* 800Hz cut-off frequency and Gain = 1 */
+/* 800Hz cut-off frequency and Gain = 0dB */
 				{0, 0, 0, 0, 0, 0, 0, 0, 0, -7554223,
 				708210, -708206, 7554225, 0, 0, 0, 0,
 				0, 0, 0, 0, 0, 6802833, -682266, 731554},
 
-/* 800Hz cut-off frequency and Gain = 0.25 */
+/* 800Hz cut-off frequency and Gain = -12dB */
 				{0, 0, 0, 0, 0, 0, 0, 0, 0, -3777112,
 				5665669, -5665667, 3777112, 0, 0, 0, 0,
 				0, 0, 0, 0, 0, 6802833, -682266, 731554},
 
-/* 800Hz cut-off frequency and Gain = 0.1 */
+/* 800Hz cut-off frequency and Gain = -20dB */
 				{0, 0, 0, 0, 0, 0, 0, 0, 0, -1510844,
 				4532536, -4532536, 1510844, 0, 0, 0, 0,
 				0, 0, 0, 0, 0, 6802833, -682266, 731554},
@@ -101,24 +101,24 @@
  * Coefficients for DL2_EQ_R
  */
 const s32 dl21_equ_coeffs[NBDL20EQ_PROFILES][NBDL2COEFFS] = {
-/* Flat response with Gain =1 */
+/* Flat response with Gain = 0dB */
 				{0, 0, 0, 0, 0,
 				0, 0, 0, 0, 0,
 				0, 0, 0x040002, 0, 0,
 				0, 0, 0, 0, 0,
 				0, 0, 0, 0, 0},
 
-/* 800Hz cut-off frequency and Gain = 1 */
+/* 800Hz cut-off frequency and Gain = 0dB */
 				{0, 0, 0, 0, 0, 0, 0, 0, 0, -7554223,
 				708210, -708206, 7554225, 0, 0, 0, 0,
 				0, 0, 0, 0, 0, 6802833, -682266, 731554},
 
-/*800Hz cut-off frequency and Gain = 0.25 */
+/*800Hz cut-off frequency and Gain = -12dB */
 				{0, 0, 0, 0, 0, 0, 0, 0, 0, -3777112,
 				5665669, -5665667, 3777112, 0, 0, 0, 0,
 				0, 0, 0, 0, 0, 6802833, -682266, 731554},
 
-/* 800Hz cut-off frequency and Gain = 0.1 */
+/* 800Hz cut-off frequency and Gain = -20dB */
 				{0, 0, 0, 0, 0, 0, 0, 0, 0, -1510844,
 				4532536, -4532536, 1510844, 0, 0, 0, 0,
 				0, 0, 0, 0, 0, 6802833, -682266, 731554},
@@ -128,21 +128,21 @@
  * Coefficients for DMICEQ
  */
 const u32 dmic_equ_coeffs[NBDMICEQ_PROFILES][NBDMICCOEFFS] = {
-/* 20kHz cut-off frequency and Gain = 1 */
+/* 20kHz cut-off frequency and Gain = 0dB */
 				{-4119413, -192384, -341428, -348088,
 				-151380, 151380, 348088, 341428, 192384,
 				4119419, 1938156, -6935719, 775202,
 				-1801934, 2997698, -3692214, 3406822,
 				-2280190, 1042982},
 
-/* 20kHz cut-off frequency and Gain = 0.25 */
+/* 20kHz cut-off frequency and Gain = -12dB */
 				{-1029873, -3078121, -5462817, -5569389,
 				-2422069, 2422071, 5569391, 5462819,
 				3078123, 1029875, 1938188, -6935811,
 				775210, -1801950, 2997722, -3692238,
 				3406838, -2280198, 1042982},
 
-/* 20kHz cut-off frequency and Gain = 0.125 */
+/* 20kHz cut-off frequency and Gain = -18dB */
 				{-514937, -1539061, -2731409, -2784693,
 				-1211033, 1211035, 2784695, 2731411,
 				1539063, 514939, 1938188, -6935811,
@@ -154,21 +154,21 @@
  * Coefficients for AMICEQ
  */
 const u32 amic_equ_coeffs[NBAMICEQ_PROFILES][NBAMICCOEFFS] = {
-/* 20kHz cut-off frequency and Gain = 1 */
+/* 20kHz cut-off frequency and Gain = 0dB */
 				{-4119413, -192384, -341428, -348088,
 				-151380, 151380, 348088, 341428, 192384,
 				4119419, 1938156, -6935719, 775202,
 				-1801934, 2997698, -3692214, 3406822,
 				-2280190, 1042982},
 
-/* 20kHz cut-off frequency and Gain = 0.25 */
+/* 20kHz cut-off frequency and Gain = -12dB */
 				{-1029873, -3078121, -5462817, -5569389,
 				-2422069, 2422071, 5569391, 5462819,
 				3078123, 1029875, 1938188, -6935811,
 				775210, -1801950, 2997722, -3692238,
 				3406838, -2280198, 1042982},
 
-/* 20kHz cut-off frequency and Gain = 0.125 */
+/* 20kHz cut-off frequency and Gain = -18dB */
 				{-514937, -1539061, -2731409, -2784693,
 				-1211033, 1211035, 2784695, 2731411,
 				1539063, 514939, 1938188, -6935811,
@@ -181,18 +181,18 @@
  * Coefficients for SDTEQ
  */
 const u32 sdt_equ_coeffs[NBSDTEQ_PROFILES][NBSDTCOEFFS] = {
-/* Flat response with Gain =1 */
+/* Flat response with Gain = 0dB */
 				{0, 0, 0, 0, 0x040002, 0, 0, 0, 0},
 
-/* 800Hz cut-off frequency and Gain = 1  */
+/* 800Hz cut-off frequency and Gain = 0dB  */
 				{0, -7554223, 708210, -708206, 7554225,
 				0, 6802833, -682266, 731554},
 
-/* 800Hz cut-off frequency and Gain = 0.25 */
+/* 800Hz cut-off frequency and Gain = -12dB */
 				{0, -3777112, 5665669, -5665667, 3777112,
 				0, 6802833, -682266, 731554},
 
-/* 800Hz cut-off frequency and Gain = 0.1 */
+/* 800Hz cut-off frequency and Gain = -20dB */
 				{0, -1510844, 4532536, -4532536, 1510844,
 				0, 6802833, -682266, 731554}
 };
Index: Kernel/sound/soc/omap/omap-abe-dsp.c
===================================================================
--- Kernel.orig/sound/soc/omap/omap-abe-dsp.c	2011-01-19 00:19:25.662071891 +0530
+++ Kernel/sound/soc/omap/omap-abe-dsp.c	2011-01-19 00:19:28.592697384 +0530
@@ -884,42 +884,42 @@
 
 static const char *dl1_equ_texts[] = {
 	"Flat response",
-	"High-pass 1dB",
-	"High-pass 0.25dB",
-	"High-pass 0.1dB",
+	"High-pass 0dB",
+	"High-pass -12dB",
+	"High-pass -20dB",
 };
 
 static const char *dl20_equ_texts[] = {
 	"Flat response",
-	"High-pass 1dB",
-	"High-pass 0.25dB",
-	"High-pass 0.1dB",
+	"High-pass 0dB",
+	"High-pass -12dB",
+	"High-pass -20dB",
 };
 
 static const char *dl21_equ_texts[] = {
 	"Flat response",
-	"High-pass 1dB",
-	"High-pass 0.25dB",
-	"High-pass 0.1dB",
+	"High-pass 0dB",
+	"High-pass -12dB",
+	"High-pass -20dB",
 };
 
 static const char *amic_equ_texts[] = {
-	"High-pass 1dB",
-	"High-pass 0.25dB",
-	"High-pass 0.125dB",
+	"High-pass 0dB",
+	"High-pass -12dB",
+	"High-pass -18dB",
 };
 
 static const char *dmic_equ_texts[] = {
-	"High-pass 1dB",
-	"High-pass 0.25dB",
-	"High-pass 0.125dB",
+	"High-pass 0dB",
+	"High-pass -12dB",
+	"High-pass -18dB",
 };
 
 static const char *sdt_equ_texts[] = {
 	"Flat response",
-	"High-pass 1dB",
-	"High-pass 0.25dB",
-	"High-pass 0.1dB",
+	"High-pass 0dB",
+	"High-pass -12dB",
+	"High-pass -20dB",
 };
 
 static const struct soc_enum dl1_equalizer_enum =
